### PR TITLE
Refactor document nature 

### LINKED ISSUE #901: Refactor: move `nature` into CoreDocument and link it directly to `frbr_uri_subtype`
Currently the `nature` FK is used only by `GenericDocument` and `LegalInstrument`. We also don't have a mechanism for faceting effectively (or controlling) the FRBR URI subtype fields for other documents, such as legislation.

I propose we combine the concept of `frbr_uri_subtype` and our explicit `DocumentNature` model into the same thing.

* [ ] make `nature` a FK field (nullable) on `CoreDocument` and remove it from `GenericDocument` and `LegalInstrument`
* [ ] enforce that `nature.code` is used for `frbr_uri_subtype`
* [ ] add `nature` as a facet to the AGP /legislation page 

This will allow us to facet all documents (in particular Legislation) by subtype. It would also help solve #638

### PR DESCRIPTION
**Pull request by actlikewill**
- [x]  Refactor document nature to core document model
- [x]  Add nature facet to legislation page

Closes #901

### CODE REVIEW

File: peachjam/migrations/0047_coredocument_new_nature.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+from itertools import chain
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+
+def move_doc_nature(apps, schema_editor):
+    Legislation = apps.get_model("peachjam", "LegalInstrument")
+    GenericDocument = apps.get_model("peachjam", "GenericDocument")
+    for doc in Legislation.objects.all().iterator(chunk_size=100):
```
Comment (longhotsummer) [inline peachjam/migrations/0047_coredocument_new_nature.py:12]:
```suggestion
    for doc in LegalInstrument.objects.all().iterator(chunk_size=100):
```

**Review by longhotsummer — COMMENTED** [review]
Looks good, just minor suggested changes to the migration.

Does the indigo adapter do the right thing if the importing document has a subtype?

nicely done.

File: peachjam/migrations/0047_coredocument_new_nature.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Generated by Django 3.2.16 on 2023-02-09 09:34
+
+from itertools import chain
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+
+def move_doc_nature(apps, schema_editor):
+    Legislation = apps.get_model("peachjam", "LegalInstrument")
```
[the remark this answers is not in the document]
Comment (actlikewill) [reply peachjam/migrations/0047_coredocument_new_nature.py]: changed

File: peachjam/migrations/0047_coredocument_new_nature.py (hunk 1 above)
Comment (actlikewill) [reply peachjam/migrations/0047_coredocument_new_nature.py]: changed

File: peachjam/adapters/adapters.py [hunk 3]
```diff
@@ -157,7 +157,7 @@ def update_document(self, url):
         model = self.get_model(document)
         logger.info(f"Importing as {model}")
 
-        if hasattr(model, "nature"):
+        if hasattr(model, "nature") and document["subtype"]:
             document_nature_name = " ".join(
```
Comment (actlikewill) [inline peachjam/adapters/adapters.py:161]: @longhotsummer I added this line because it seems some documents come in without the subtype property

Comment (longhotsummer) [reply peachjam/adapters/adapters.py:161]: The model now will always have `nature`, so that can be skipped. Guarding against a subtype makes sense.

**Review by longhotsummer — DISMISSED** [review]
Very minor change, if you want to make it.

**Review by longhotsummer — APPROVED** [review]

**Pull request merged**